reference-implementation/lib/exclusive-stream-reader.js

```diff
@@ -89,11 +79,11 @@ export default class ExclusiveStreamReader {
     }
 
     if (this._encapsulatedReadableStream._readableStreamReader !== this) {
-      return this._closedAfterRelease;
+      return this._closedPromise;
     }
 
-    this.releaseLock();
-    return this._encapsulatedReadableStream.cancel(reason, ...args);
+    // Bypass lock check.
+    return CancelReadableStream(this._encapsulatedReadableStream, reason, ...args);
```

Comment:
I think it is OK to stop passing any args now since it is not a direct delegation to a possibly-overridden `.cancel` method.